std/string.d

```diff
@@ -221,10 +221,52 @@ inout(char)[] fromStringz(inout(char)* cString) @nogc @system pure nothrow {
 ///
 @system pure unittest
 {
-    assert(fromStringz(null) == null);
+    char* foo = null;
+    assert(fromStringz(foo) == null);
     assert(fromStringz("foo") == "foo");
 }
 
+/++
+    Params:
+        cString = A null-terminated c-style 16bit wide string.
+
+    Returns: A D-style array of `wchar` referencing the same string.  The
+    returned array will retain the same type qualifiers as the input.
+
+    $(RED Important Note:) The returned array is a slice of the original buffer.
+    The original data is not changed and not copied.
++/
+
+inout(wchar)[] fromStringz(inout(wchar)* cString) @nogc @system pure nothrow
+{
+    version (Windows)
+    {
+        import core.stdc.wchar_ : wcslen;
+
+        alias strlen16 = wcslen;
+    }
+    else version (Posix)
+    {
+        size_t strlen16(const wchar* s)
+        {
+            const(wchar)* p = s;
+            while (*p)
+                p++;
+            return p - s;
+        }
```

Comment:
My personal opinion: Correctness (and no potential weird corner-cases) > performance (and I doubt that this can be optimized much further by the stdc implementation)

Reply:
Ok I agree, but as an aside, there is this: https://code.woboq.org/userspace/glibc/wcsmbs/wcslen.c.html  Is this faster? If so I don't really understand why ;)

Reply:
It is manual loop unrolling.

Reply:
This is even more fun: http://www.stdlib.net/~colmmacc/strlen.c.html (not for wide string though)